WalletWasabi/WabiSabi/Client/RoundStateUpdater.cs

```diff
@@ -4,18 +4,29 @@
 using System.Threading;
 using System.Threading.Tasks;
 using WalletWasabi.Bases;
+using WalletWasabi.Tor.Http;
+using WalletWasabi.Tor.Socks5.Pool.Circuits;
 using WalletWasabi.WabiSabi.Backend.PostRequests;
 using WalletWasabi.WabiSabi.Models;
+using WalletWasabi.WebClients.Wasabi;
 
 namespace WalletWasabi.WabiSabi.Client
 {
 	public class RoundStateUpdater : PeriodicRunner
 	{
+		private readonly PersonCircuit? _personCircuit;
+
 		public RoundStateUpdater(TimeSpan requestInterval, IWabiSabiApiRequestHandler arenaRequestHandler) : base(requestInterval)
 		{
 			ArenaRequestHandler = arenaRequestHandler;
 		}
 
+		public RoundStateUpdater(TimeSpan requestInterval, IWasabiHttpClientFactory backendHttpClientFactory) : base(requestInterval)
+		{
+			_personCircuit = backendHttpClientFactory.NewHttpClientWithPersonCircuit(out IHttpClient httpClient);
+			ArenaRequestHandler = new WabiSabiHttpApiClient(httpClient);
+		}
```

Comment:
Why building the request handler here when it can be received as before? This introduces an extra dependency that shoudn't be here. Please do not add this because it has been extremely hard to keep the design decoupled.